doc/nrf/config_and_build/hwmv2_migration.rst

```diff
@@ -0,0 +1,57 @@
+.. _hwmv1_to_v2_migration:
+
+Migrating to the current hardware model
+#######################################
+
+Starting with |NCS| version 2.7.0, the hardware model, which is to say the
+manner in which SoCs and boards are named, defined and used, has been overhauled. This
+section describes how to migrate from the hardware model used in |NCS| versions
+2.6.0 and earlier to the current one.
+
+::note The current hardware model was introduced shortly after the relase of
+3.6.0 in the Zephyr Project.
+
+The current hardware model adds support for features that had been identified as important
+over the years. Among them:
+
+- Support for multi-core, multi-arch AMP (Asymmetrical Multi Processing) SoCs
+- Support for multi-SoC boards
+- Support for reusing the SoC and board Kconfig trees outside of the Zephyr and
+  NCS build system
+- Support for advanced use cases with :ref:`zephyr:sysbuild`
+- Removal of all existing arbitrary and inconsistent uses of Kconfig and folder
+  names
+
+More information about the rationale, development and concepts behind the new
+model can be found in the `Zephyr original issue <51831>`_, the
+`Zephyr original Pull Request <50305>`_ and, for a complete set of changes
+introduced, the `hardware model v2 commit`_.
+
+Some non-critical features, enhancements and improvements of the new hardware
+model are still in development. Check the
+`Zephyr hardware model v2 enhancements issue <69546>`_ for a complete list.
+
+For |NCS| users that also have out-of-tree boards, the transition from the previous hardware model to the current one (commonly
+referred to as "hardware model v2") is optional, because |NCS| provides
+full backwards compatibility for any existing out-of-tree boards.
```

Comment:
```suggestion
For |NCS| users that also have out-of-tree boards, the transition from hardware model v1 to hardware model v2 is optional, because |NCS| provides full backwards compatibility for any existing out-of-tree boards.
```